src/connectdlg.h

```diff
@@ -59,12 +59,27 @@ class CConnectDlg : public CBaseDlg, private Ui_CConnectDlgBase
     void SetConnClientsList ( const CHostAddress& InetAddr, const CVector<CChannelInfo>& vecChanInfo );
 
     void SetPingTimeAndNumClientsResult ( const CHostAddress& InetAddr, const int iPingTime, const int iNumClients );
+    void SetServerVersionResult ( const CHostAddress& InetAddr, const QString& strVersion );
 
     bool    GetServerListItemWasChosen() const { return bServerListItemWasChosen; }
     QString GetSelectedAddress() const { return strSelectedAddress; }
     QString GetSelectedServerName() const { return strSelectedServerName; }
 
 protected:
+    // NOTE: This enum must list the columns in the same order
+    //       as their column headings in connectdlgbase.ui
+    enum EConnectListViewColumns
+    {
+        LVC_NAME,               // server name
+        LVC_PING,               // ping time
+        LVC_CLIENTS,            // number of connected clients (including additional strings like " (full)")
+        LVC_LOCATION,           // location
+        LVC_VERSION,            // server version
+        LVC_PING_MIN_HIDDEN,    // minimum ping time (invisible)
+        LVC_CLIENTS_MAX_HIDDEN, // maximum number of clients (invisible)
+        LVC_COLUMNS             // total number of columns
```

Comment:
Alternatively we could most likely also calculate this programmatically. It could go out of sync in future due to a mistake?

Reply:
I think a better approach would be to remove the headings from the widget in the `.ui` file and to set them in the code in the same section as we set the widths. I did consider this, but didn't get around to looking up the correct way to do so. Then the enum would be the one source of truth for column order.